https://app.circleci.com/pipelines/github/checkstyle/checkstyle/48954/workflows/8ec1aa8a-dd4a-4624-838f-ba5de7dce157/jobs/1691538

```
    [WARNING] Generating /home/circleci/project/target/apidocs/com/puppycrawl/tools/checkstyle/checks/javadoc/InvalidJavadocTag.html...
    [WARNING] /home/circleci/project/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/InvalidJavadocTag.java:42: warning: no @return
    [WARNING] public int getLine() {
    [WARNING] ^
    [WARNING] /home/circleci/project/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/InvalidJavadocTag.java:47: warning: no @return
    [WARNING] public int getCol() {
    [WARNING] ^
    [WARNING] /home/circleci/project/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/InvalidJavadocTag.java:52: warning: no @return
    [WARNING] public String getName() {
    [WARNING] ^


    [WARNING] Generating /home/circleci/project/target/apidocs/com/puppycrawl/tools/checkstyle/grammar/SimpleToken.html...
    [WARNING] /home/circleci/project/src/main/java/com/puppycrawl/tools/checkstyle/grammar/SimpleToken.java:38: warning: no comment
    [WARNING] private static final long serialVersionUID = 1L;
    [WARNING] ^



    [WARNING] Generating /home/circleci/project/target/apidocs/com/puppycrawl/tools/checkstyle/checks/javadoc/utils/TagInfo.html...
    [WARNING] /home/circleci/project/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/utils/TagInfo.java:44: warning: no @return
    [WARNING] public String getName() {
    [WARNING] ^
    [WARNING] /home/circleci/project/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/utils/TagInfo.java:49: warning: no @return
    [WARNING] public String getValue() {
    [WARNING] ^
    [WARNING] /home/circleci/project/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/utils/TagInfo.java:54: warning: no @return
    [WARNING] public LineColumn getPosition() {
    [WARNING] ^


    [WARNING] 7 warnings
    [INFO] Building jar: /home/circleci/project/target/checkstyle-13.9.0-SNAPSHOT-javadoc.jar
```

--- a/.ci/validation.sh
+++ b/.ci/validation.sh
@@ -1589,6 +1589,10 @@ spotless)
   ./mvnw -e --no-transfer-progress spotless:check
   ;;
 
+javadoc)
+  ./mvnw -e --no-transfer-progress javadoc:javadoc
+  ;;
+
 openrewrite-checkstyle-auto-fix)
   echo "Cloning and building OpenRewrite recipes..."
   PROJECT_ROOT="$(pwd)"

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -227,6 +227,10 @@ workflows:
           resource_class: medium+
           command: ./.ci/validation.sh spotbugs-and-pmd
       - validate-with-maven-script:
+          name: javadoc
+          image-name: *cs_img
+          command: ./.ci/validation.sh javadoc
+      - validate-with-maven-script:
           name: site
           image-name: *cs_img
           command: ./.ci/validation.sh site

--- a/pom.xml
+++ b/pom.xml
@@ -613,7 +613,7 @@
             <show>private</show>
             <failOnError>true</failOnError>
             <failOnWarnings>true</failOnWarnings>
-            <doclint>-missing</doclint>
+            <doclint>all</doclint>
             <linksource>true</linksource>
             <tags>
               <tag>

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/InvalidJavadocTag.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/InvalidJavadocTag.java
@@ -28,17 +28,29 @@ package com.puppycrawl.tools.checkstyle.checks.javadoc;
  */
 public record InvalidJavadocTag(int line, int col, String name) {
 
-    /** Legacy getter for line (backward compatibility). */
+    /**
+     * Legacy getter for line (backward compatibility).
+     *
+     * @return the line of the invalid tag
+     */
     public int getLine() {
         return line;
     }
 
-    /** Legacy getter for col (backward compatibility). */
+    /**
+     * Legacy getter for col (backward compatibility).
+     *
+     * @return the column of the invalid tag
+     */
     public int getCol() {
         return col;
     }
 
-    /** Legacy getter for name (backward compatibility). */
+    /**
+     * Legacy getter for name (backward compatibility).
+     *
+     * @return the name of the invalid tag
+     */
     public String getName() {
         return name;
     }

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/utils/TagInfo.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/utils/TagInfo.java
@@ -30,17 +30,29 @@ import com.puppycrawl.tools.checkstyle.api.LineColumn;
  */
 public record TagInfo(String name, String value, LineColumn position) {
 
-    /** Legacy getter for tag name (backward compatibility). */
+    /**
+     * Legacy getter for tag name (backward compatibility).
+     *
+     * @return the name of the tag
+     */
     public String getName() {
         return name;
     }
 
-    /** Legacy getter for tag value (backward compatibility). */
+    /**
+     * Legacy getter for tag value (backward compatibility).
+     *
+     * @return the value of the tag
+     */
     public String getValue() {
         return value;
     }
 
-    /** Legacy getter for tag position (backward compatibility). */
+    /**
+     * Legacy getter for tag position (backward compatibility).
+     *
+     * @return the position of the tag
+     */
     public LineColumn getPosition() {
         return position;
     }

--- a/src/main/java/com/puppycrawl/tools/checkstyle/grammar/SimpleToken.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/grammar/SimpleToken.java
@@ -34,6 +34,7 @@ import org.antlr.v4.runtime.Token;
  */
 public final class SimpleToken extends CommonToken {
 
+    /** A unique serial version identifier. */
     @Serial
     private static final long serialVersionUID = 1L;
 

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTagTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTagTest.java
@@ -24,6 +24,8 @@ import static com.google.common.truth.Truth.assertWithMessage;
 
 import org.junit.jupiter.api.Test;
 
+import com.puppycrawl.tools.checkstyle.api.LineColumn;
+import com.puppycrawl.tools.checkstyle.checks.javadoc.utils.TagInfo;
 import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
 
 public class JavadocTagTest {
@@ -92,6 +94,53 @@ public class JavadocTagTest {
         assertThat(new JavadocTag(0, 0, "value", null).isInlineTag()).isTrue();
         assertThat(new JavadocTag(0, 0, "see", null).isInlineTag()).isFalse();
 
+    }
+
+    @Test
+    public void testInvalidJavadocTagLegacyGetters() {
+        final InvalidJavadocTag tag = new InvalidJavadocTag(1, 2, "test");
+        assertWithMessage("Invalid line number")
+            .that(tag.getLine())
+            .isEqualTo(1);
+        assertWithMessage("Invalid column number")
+            .that(tag.getCol())
+            .isEqualTo(2);
+        assertWithMessage("Invalid tag name")
+            .that(tag.getName())
+            .isEqualTo("test");
+        assertWithMessage("Invalid line number")
+            .that(tag.line())
+            .isEqualTo(1);
+        assertWithMessage("Invalid column number")
+            .that(tag.col())
+            .isEqualTo(2);
+        assertWithMessage("Invalid tag name")
+            .that(tag.name())
+            .isEqualTo("test");
+    }
+
+    @Test
+    public void testTagInfoLegacyGetters() {
+        final LineColumn position = new LineColumn(1, 2);
+        final TagInfo tagInfo = new TagInfo("name", "value", position);
+        assertWithMessage("Invalid name")
+            .that(tagInfo.getName())
+            .isEqualTo("name");
+        assertWithMessage("Invalid value")
+            .that(tagInfo.getValue())
+            .isEqualTo("value");
+        assertWithMessage("Invalid position")
+            .that(tagInfo.getPosition())
+            .isEqualTo(position);
+        assertWithMessage("Invalid name")
+            .that(tagInfo.name())
+            .isEqualTo("name");
+        assertWithMessage("Invalid value")
+            .that(tagInfo.value())
+            .isEqualTo("value");
+        assertWithMessage("Invalid position")
+            .that(tagInfo.position())
+            .isEqualTo(position);
     }
 
 }

